static/sidebar/js/mobile/modal/index.js

```diff
@@ -151,7 +160,9 @@
             data.runTime.loginid || data.runTime.loginId,
             function(data){
               setTimeout(function(){
+                checkProducts(data);
```

Comment:
What if we check for the products data length here, and then we just add a class `tvp-no-products` to the top most element, then with CSS we can hide the elements we need to hide, for example:

```
if (!data || !data.length) {
 el.classList.add('tvp-no-products');
}
```

Then, in the CSS side you can have:

```
.tvp-no-products .tvp-related-products {
  display: none;
}

.tvp-no-products .tvp-products {
  display: none;
}
```